src/OrbitQt/orbitmainwindow.cpp

```diff
@@ -1169,35 +1169,28 @@ void OrbitMainWindow::LoadCaptureOptionsIntoApp() {
   app_->SetEnableApi(settings.value(kEnableApiSettingKey, true).toBool());
   app_->SetEnableIntrospection(settings.value(kEnableIntrospectionSettingKey, false).toBool());
 
-  if (absl::GetFlag(FLAGS_tracepoint_callstack_collection)) {
-    CaptureOptions::ThreadStateChangeCallStackCollection const
-        collect_callstack_on_thread_state_change =
-            static_cast<CaptureOptions::ThreadStateChangeCallStackCollection>(
-                settings
-                    .value(kEnableCallStackCollectionOnThreadStateChanges,
-                           orbit_qt::CaptureOptionsDialog::
-                               kThreadStateChangeCallStackCollectionDefaultValue)
-                    .toInt());
-    if (settings.value(kCollectThreadStatesSettingKey, false).toBool()) {
-      app_->SetThreadStateChangeCallstackCollection(collect_callstack_on_thread_state_change);
-    } else {
-      app_->SetThreadStateChangeCallstackCollection(
-          CaptureOptions::kThreadStateChangeCallStackCollectionUnspecified);
-    }
-
-    uint16_t stack_dump_size = static_cast<uint16_t>(
-        settings
-            .value(
-                kThreadStateChangeCallstackMaxCopyRawStackSizeSettingKey,
-                orbit_qt::CaptureOptionsDialog::kThreadStateChangeMaxCopyRawStackSizeDefaultValue)
-            .toUInt());
-    app_->SetThreadStateChangeCallstackStackDumpSize(stack_dump_size);
+  CaptureOptions::ThreadStateChangeCallStackCollection const
+      collect_callstack_on_thread_state_change = static_cast<
+          CaptureOptions::ThreadStateChangeCallStackCollection>(
+          settings
+              .value(
+                  kEnableCallStackCollectionOnThreadStateChanges,
+                  orbit_qt::CaptureOptionsDialog::kThreadStateChangeCallStackCollectionDefaultValue)
+              .toInt());
+  if (settings.value(kCollectThreadStatesSettingKey, false).toBool()) {
+    app_->SetThreadStateChangeCallstackCollection(collect_callstack_on_thread_state_change);
   } else {
     app_->SetThreadStateChangeCallstackCollection(
         CaptureOptions::kThreadStateChangeCallStackCollectionUnspecified);
-
-    app_->SetThreadStateChangeCallstackStackDumpSize(std::numeric_limits<uint16_t>::max());
   }
+
+  uint16_t stack_dump_size = static_cast<uint16_t>(
+      settings
+          .value(kThreadStateChangeCallstackMaxCopyRawStackSizeSettingKey,
+                 orbit_qt::CaptureOptionsDialog::kThreadStateChangeMaxCopyRawStackSizeDefaultValue)
+          .toUInt());
+  app_->SetThreadStateChangeCallstackStackDumpSize(stack_dump_size);
```

Comment:
We were setting the stack_dump_size as max before in case of unspecified callstack collection. I guess that this doesn't matter at all in this case, since it will be ignored, but could we have a potential issue because unespecified value?

Reply:
Yeah, that shouldn't be an issue. That value is ignored in that case anyways.